lib/nrf5/common/uarte_common.c

```diff
@@ -0,0 +1,429 @@
+{
+	return UARTE_RXD_MAXCNT(uarte);	
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief UARTE Send Data Word.
+
+Setup Data word, reset events, start transmission.
+
+@param[in] uarte unsigned 32 bit. UARTE block register address base @ref
+uarte_reg_base
+@param[in] *ptr unsigned 8 bit. Address pointer of data which be transmitted
+@param[in] maxcnt unsigned 8 bit. Maximum number of bytes in transmit buffer
+*/
+
+void uarte_send(uint32_t uarte, uint8_t *ptr, uint8_t maxcnt)
+{
+	uarte_tx_packet(uarte, ptr, maxcnt);
+	uarte_tx_init(uarte);
+	uarte_tx_start(uarte);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief UARTE Send Data Word with Blocking.
+
+Setup Data word, reset events, start trasmission and block until all bytes
+in the TXD buffer transmitted.
+
+@param[in] uarte unsigned 32 bit. UARTE block register address base @ref
+uarte_reg_base
+@param[in] *ptr unsigned 8 bit. Address pointer of data which be transmitted
+@param[in] maxcnt unsigned 8 bit. Maximum number of bytes in transmit buffer
+*/
+
+void uarte_send_waiting(uint32_t uarte, uint8_t *ptr, uint8_t maxcnt)
+{
+	uarte_tx_packet(uarte, ptr, maxcnt);
+	uarte_tx_init(uarte);
+	uarte_tx_start(uarte);
+	while(UARTE_EVENTS_ENDTX(uarte) == 0);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief UARTE interrupt enable.
+
+The events can be selected as CTS, NCTS, RXDRDY, ENDRX, TXDRDY, ENDTX, ERROR,
+RXTO, RXSTARTED, TXSTARTED, and/or TXSTOPPED.
+
+@param[in] uarte unsigned 32 bit. UARTE block register address base @ref
+uarte_reg_base
+@param[in] events unsigned 32 bit. predetermined event names @ref 
+uarte_events_interrupt
+*/
+
+void uarte_interrupt_enable(uint32_t uarte, uint32_t events)
+{
+	UARTE_INTENSET(uarte) |= events;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief UARTE interrupt disable.
+
+The events can be selected as CTS, NCTS, RXDRDY, ENDRX, TXDRDY, ENDTX, ERROR,
+RXTO, RXSTARTED, TXSTARTED, and/or TXSTOPPED.
+
+@param[in] uarte unsigned 32 bit. UARTE block register address base @ref
+uarte_reg_base
+@param[in] events unsigned 32 bit. predetermined event names @ref 
+uarte_events_interrupt
+*/
+
+void uarte_interrupt_disable(uint32_t uarte, uint32_t events)
+{
+	UARTE_INTENCLR(uarte) |= events;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief UARTE Get Events.
+
+The events can be selected as CTS, NCTS, RXDRDY, ENDRX, TXDRDY, ENDTX, ERROR,
+RXTO, RXSTARTED, TXSTARTED, or TXSTOPPED.
+
+@param[in] uarte unsigned 32 bit. UARTE block register address base @ref
+uarte_reg_base
+@param[in] event unsigned 32 bit. predetermined event names @ref 
+uarte_events_interrupt
+@returns boolean
+*/
+
+bool uarte_get_event(uint32_t uarte, char event)
+{
+	if(event & CTS){return (UARTE_EVENTS_CTS(uarte) != 0);}
```

Comment:
stylecheck here.

Reply:
I dont get stylecheck here, but this may because of "return false". It will be removed.